### PR TITLE
feat: [#449] Implement HTTP response stream

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

Closes https://github.com/goravel/goravel/issues/449

@coderabbitai summary

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: context_response_test.go [hunk 1]
```diff
@@ -701,3 +702,86 @@ func TestResponse_Status(t *testing.T) {
...
+	var (
+		err        error
+		fiber      *Route
+		req        *http.Request
+		mockConfig *configmocks.Config
+	)
+
+	beforeEach := func() {
+		mockConfig = &configmocks.Config{}
+		mockConfig.On("GetBool", "http.drivers.fiber.prefork", false).Return(false).Once()
```
Comment (hwbrzzl) [inline context_response_test.go:716]:
```suggestion
		mockConfig.EXCEPT().GetBool("http.drivers.fiber.prefork", false).Return(false).Once()
```

File: context_response_test.go [hunk 2]
```diff
@@ -701,3 +702,86 @@ func TestResponse_Status(t *testing.T) {
...
+	}
+
+	tests := []struct {
+		name   string
+		method string
+		url    string
+		setup  func(method, url string) error
+		output []string
+	}{
+		{
```
Comment (hwbrzzl) [inline context_response_test.go:729]: If there is only one test case, we don't need to use the `for` structure.

File: context_response_test.go [hunk 3]
```diff
@@ -701,3 +702,86 @@ func TestResponse_Status(t *testing.T) {
...
+			name:   "Stream",
+			method: "GET",
+			url:    "/stream",
+			setup: func(method, url string) error {
+				fiber.Get("/stream", func(ctx contractshttp.Context) contractshttp.Response {
+					return ctx.Response().Status(http.StatusCreated).Stream(func(w contractshttp.StreamWriter) error {
+						b := []string{"a", "b", "c", "c", "c", "c", "c"}
```
Comment (hwbrzzl) [inline context_response_test.go:736]:
```suggestion
						b := []string{"a", "b", "c"}
```

File: context_response_test.go [hunk 4]
```diff
@@ -701,3 +702,86 @@ func TestResponse_Status(t *testing.T) {
...
+			assert.Nil(t, err)
+
+			resp, err := fiber.Test(req)
+			assert.NoError(t, err)
+
+			scanner := bufio.NewScanner(resp.Body)
+			var output []string
+			for scanner.Scan() {
+				output = append(output, scanner.Text())
+			}
```
Comment (hwbrzzl) [inline context_response_test.go:779]: I doubt that this way can't test the stream accurately. I asked GPT, could you try the code below?

```
package main

import (
    "time"

    "github.com/gofiber/fiber/v2"
)

func streamData(c *fiber.Ctx) error {
    c.Set(fiber.HeaderContentType, fiber.MIMETextHTMLCharsetUTF8)

    streamErr := c.Context().SetBodyStreamWriter(func(w *fiber.StreamWriter) error {
        for i := 0; i < 10; i++ {
            _, err := w.Write([]byte("Streaming data " + string(i+48) + "..."))
            if err != nil {
                return err
            }
            time.Sleep(100 * time.Millisecond) // Simulate delay between chunks
        }
        return nil
    })

    if streamErr != nil {
        return streamErr
    }

    return nil
}

func main() {
    app := fiber.New()

    app.Get("/stream", streamData)

    app.Listen(":3000")
}
```

```
package main

import (
    "bufio"
    "net/http/httptest"
    "strings"
    "testing"
    "time"

    "github.com/gofiber/fiber/v2"
    "github.com/stretchr/testify/assert"
)

func TestStreamData(t *testing.T) {
    app := fiber.New()

    app.Get("/stream", streamData)

    req := httptest.NewRequest("GET", "/stream", nil)
    resp, err := app.Test(req, -1) // -1 means read until we get the full response or the connection is closed

    assert.NoError(t, err)
    assert.Equal(t, 200, resp.StatusCode)
    assert.Equal(t, resp.Header.Get("Content-Type"), fiber.MIMETextHTMLCharsetUTF8)

    expectedChunks := []string{
        "Streaming data 0...",
        "Streaming data 1...",
        "Streaming data 2...",
        "Streaming data 3...",
        "Streaming data 4...",
        "Streaming data 5...",
        "Streaming data 6...",
        "Streaming data 7...",
        "Streaming data 8...",
        "Streaming data 9...",
    }

    reader := bufio.NewReader(resp.Body)
    for i, expected := range expectedChunks {
        chunk, err := reader.ReadString(' ')
        if err != nil {
            t.Fatalf("Failed to read chunk: %v", err)
        }
        chunk += ".." // Append rest of the string
        rest, _ := reader.ReadString('.')
        chunk += rest

        assert.Equal(t, expected, strings.TrimSuffix(chunk, "\n"), "Mismatch in chunk %d", i+1)
        // To simulate reading it with an interval as in real-time
        time.Sleep(100 * time.Millisecond)
    }
}
```

Comment (kkumar-gcc) [reply context_response_test.go:779]: Both are similar

Comment (hwbrzzl) [reply context_response_test.go:779]: I tested them, and yes, both of them get the final response(a,b,c), then scan the result, this way can't test the output data verbatim. It's not the perfect solution, but I think it's good enough for us now. Great job.

**Review by hwbrzzl — APPROVED** [review]

File: context_response_test.go [hunk 5]
```diff
@@ -701,3 +702,60 @@ func TestResponse_Status(t *testing.T) {
...
+
+	beforeEach := func() {
+		mockConfig = &configmocks.Config{}
+		mockConfig.EXPECT().GetBool("http.drivers.fiber.prefork", false).Return(false).Once()
+		mockConfig.EXPECT().GetInt("http.drivers.fiber.body_limit", 4096).Return(4096).Once()
+		mockConfig.EXPECT().GetInt("http.drivers.fiber.header_limit", 4096).Return(4096).Once()
+		ConfigFacade = mockConfig
+	}
+
+	t.Run("Stream", func(t *testing.T) {
```
Comment (hwbrzzl) [inline context_response_test.go:722]: Actually, we don't need this line and `beforeEach` in this case anymore, you can write the single test case directly.

**Review by hwbrzzl — APPROVED** [review]
Great PR 👍

**Pull request merged**